keycloak-sms-provider-aws-sns/src/main/java/cc/coopersoft/keycloak/phone/providers/sender/AwsSnsSmsSenderService.java

```diff
@@ -0,0 +1,64 @@
+package cc.coopersoft.keycloak.phone.providers.sender;
+
+import cc.coopersoft.keycloak.phone.providers.exception.MessageSendException;
+import cc.coopersoft.keycloak.phone.providers.spi.FullSmsSenderAbstractService;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.AmazonSNSClientBuilder;
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AwsSnsSmsSenderService extends FullSmsSenderAbstractService {
+
+    private static final Logger logger = Logger.getLogger(AwsSnsSmsSenderService.class);
+    private final Config.Scope config;
+
+    public AwsSnsSmsSenderService(String realmDisplay, Config.Scope config) {
+        super(realmDisplay);
+        this.config = config;
+    }
+
+    @Override
+    public void sendMessage(String phoneNumber, String message) throws MessageSendException {
+        AmazonSNS snsClient = AmazonSNSClientBuilder.standard().build();
+        String senderId = config.get("SENDER_ID");
+
+        Map<String, MessageAttributeValue> smsAttributes = null;
+
+        if (senderId != null) {
+            smsAttributes = new HashMap<>();
+            smsAttributes.put("AWS.SNS.SMS.SenderID", new MessageAttributeValue()
+                    .withStringValue(senderId)
+                    .withDataType("String"));
+        }
+
+        logger.debug(String.format("Sending phone verification code via aws sns: %s >>> %s", phoneNumber, message));
+
+        try {
+            sendSMSMessage(snsClient, message, phoneNumber, smsAttributes);
+        } catch (Exception ex) {
+            String msg = "Could not send message via aws sns";
+            logger.error(msg, ex);
+            throw new MessageSendException(msg, ex);
+        }
+    }
+
+    private static void sendSMSMessage(AmazonSNS snsClient, String message,
+                                       String phoneNumber, Map<String, MessageAttributeValue> smsAttributes) {
+        PublishResult result = snsClient.publish(new PublishRequest()
+                .withMessage(message)
+                .withPhoneNumber(phoneNumber)
+                .withMessageAttributes(smsAttributes));
+
+        logger.debug(String.format("Sent phone verification code via aws sns: %s >>> %s with message id %s", phoneNumber, message, result.getMessageId()));
```

Comment:
@dominikschopf : better to not log here because of sensitive data. If we still want to log it, i would recommend  to move the statement to the generic part of the plugin in order to do it for all sender implementation.

Reply:
@towe75 I removed the information from the log outputs. The phone number is still printed in `TokenCodeResource.getTokenCode`.